app/goal_routes.py

```diff
@@ -0,0 +1,74 @@
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+from .route_helpers import validate_model
+from flask import Blueprint, jsonify, request
+
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+@goals_bp.route("", methods=["POST"])
+# Creates a new goal and returns it as json
+def create_new_goal():
+    request_body = request.get_json()
+    if "title" in request_body:
+        new_goal = Goal.from_dict(request_body)
+
+        db.session.add(new_goal)
+        db.session.commit()
+        return {"goal": new_goal.as_dict()}, 201
+    else:
+        return {"details": "Invalid data"}, 400
```

Comment:
It's good that you're doing data validation, but you should indicate what data is invalid.